### PR TITLE
feat(SWHagent): add status of request to DB

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
## Description

Store status of software heritage results in database.

### Changes

- Remove SW Heritage agent from calculating ETA.
- Add All licenses found through SW Heritage to a single column in DB.

## How to test

* Test SW Heritage agent end to end.

### CODE REVIEW

File: src/softwareHeritage/agent/softwareHeritageAgent.php [hunk 1]
```diff
@@ -162,16 +163,21 @@ protected function getSoftwareHeritageLicense($sha256)
...
+    $cookedResult = array();
+    switch ($statusCode) :
+      case SoftwareHeritageDao::SWH_STATUS_OK:
+        $responseContent = json_decode($response->getBody()->getContents(),true);
+        $cookedResult = $responseContent["facts"][0]["licenses"];
+        return array(SoftwareHeritageDao::SWH_STATUS_OK, $cookedResult);
+      case SoftwareHeritageDao::SWH_BAD_REQUEST:
+        return array(SoftwareHeritageDao::SWH_BAD_REQUEST, $cookedResult);
+      case SoftwareHeritageDao::SWH_NOT_FOUND:
+        return array(SoftwareHeritageDao::SWH_NOT_FOUND, $cookedResult);
```
Comment (GMishx) [inline src/softwareHeritage/agent/softwareHeritageAgent.php]: Suggesting to reduce the code to something like:

```php
      case SoftwareHeritageDao::SWH_STATUS_OK:
        $responseContent = json_decode($response->getBody()->getContents(),true);
        $cookedResult = $responseContent["facts"][0]["licenses"];
      case SoftwareHeritageDao::SWH_BAD_REQUEST:
      case SoftwareHeritageDao::SWH_NOT_FOUND:
        return array($statusCode, $cookedResult);
```

File: src/softwareHeritage/agent/softwareHeritageAgent.php [hunk 2]
```diff
@@ -162,16 +163,21 @@ protected function getSoftwareHeritageLicense($sha256)
...
-      $licenseRecord = $responseContent["facts"][0]["licenses"];
-      return $licenseRecord;
-    } else if (429 === $statusCode) {
-      $responseContent = $response->getHeaders();
-      return $responseContent["X-RateLimit-Reset"][0];
-    } else {
-      return [];
-    }
+    $cookedResult = array();
+    switch ($statusCode) :
```
Comment (GMishx) [inline src/softwareHeritage/agent/softwareHeritageAgent.php]: This `switch/case` syntax is not familiar to me. Can you please provide an reference to this syntax?

**Comment by GMishx** [comment]
Please fix the error in Travis:
```
PHP Warning:  pg_query(): Query failed: ERROR:  column "status" of relation "software_heritage" does not exist in /home/travis/build/fossology/fossology/src/lib/php/Db/Driver/Postgres.php on line 79
[2020-02-18 14:04:55] default.CRITICAL: ERROR:  column "status" of relation "software_heritage" does not exist [] []
PHP Fatal error:  Uncaught Fossology\Lib\Exception: error executing: ALTER TABLE "software_heritage" ALTER COLUMN "swh_status" DROP NOT NULL, ALTER COLUMN "status" SET DEFAULT 1
ERROR:  column "status" of relation "software_heritage" does not exist in /home/travis/build/fossology/fossology/src/lib/php/Db/DbManager.php:144

Stack trace:
#0 /home/travis/build/fossology/fossology/src/lib/php/Db/DbManager.php(203): Fossology\Lib\Db\DbManager->checkResult(false, 'ALTER TABLE "so...')
#1 /home/travis/build/fossology/fossology/src/lib/php/Test/TestAbstractDb.php(153): Fossology\Lib\Db\DbManager->queryOnce('ALTER TABLE "so...')
#2 /home/travis/build/fossology/fossology/src/testing/db/createTestEnvironment.php(47): Fossology\Lib\Test\TestAbstractDb->alterTables(Array, true)
#3 {main}
  thrown in /home/travis/build/fossology/fossology/src/lib/php/Db/DbManager.php on line 144
```

Maybe you forgot the change the `core-schema.dat`?

**Comment by shaheemazmalmmd** [comment]
@GMishx : i think i missed it when i pushed. let me fix it quickly.

**Comment by ag4ums** [comment]
SH status seems to be incorrect :  
image

* File exists with SH and has license : status showing red
`https://archive.softwareheritage.org/api/1/content/sha256:6fcd5f40b823ad1ebb0582cdc51dd2c745faf4ac022f45f3b03687f380d103c0`
`https://archive.softwareheritage.org/api/1/content/sha256:4eb6dc7b63e6823ff3c68bf4c3209999d98fdaf7ddd4e59961d061134c86f8f6`

* File exist with SH and no license : status showing green
`https://archive.softwareheritage.org/api/1/content/sha256:f72500759b3a5faba0f66d49b302e1e3df7b1a1f17e27f74ea1aa98499a56fe5`

**Comment by shaheemazmalmmd** [comment]
@ag4ums : i have just tested it again with some files of fckeditor. it looks fine to me. please see the attached screenshot.

screenShot6

Can you please share the file with which you have tested ?

**Comment by ag4ums** [comment]
@shaheemazmalmmd try to find the files with the same hash value

**Comment by shaheemazmalmmd** [comment]
I have tested it locally and it looks fine to me. please find the attached screenshot.

screenShot7

**Comment by ag4ums** [comment]
may be there are some setup issue, rescan worked fine.

**Pull request merged**